bitnami/zookeeper/values.yaml

```diff
@@ -766,6 +775,12 @@ tls:
     ## @param tls.client.passwordsSecretName Existing secret containing Keystore and truststore passwords
     ##
     passwordsSecretName: ""
+    ## @param tls.client.keystorePasswordSecretKey The secret key from the passwordsSecretName if 'keystorePasswordSecretKey' key different from the default (keystore-password)
+    ##
+    keystorePasswordSecretKey: "keystore-password"
+    ## @param tls.client.truststorePasswordSecretKey The secret key from the passwordsSecretName if 'truststorePasswordSecretKey' key different from the default (truststore-password)
+    ##
+    truststorePasswordSecretKey: "truststore-password"
```

Comment:
>By the way, note that having the possibility to set both keystore-password and truststore-password for zookeeper can lead to a mismatch error because zookeeper REQUIRES the same keystore and truststore password. But this is another debate.

I am not really sure about this, in fact, I believe that this chart is setting two different random passwords if the TLS configuration is autogenerated instead of provided:

https://github.com/bitnami/charts/blob/7fb947bbd62c1e5d4a91b0378d65c66cf0c672c0/bitnami/zookeeper/templates/secrets.yaml#L55-L56

> I think this approach would be true if cert-manager allows a different password for keystore and for truststore. But to the best of my knowledge, we can configure only one secret through passwordSecretRef with only one key to set both passwords, so we have to share the same password

You are right! But then, I believe this is really a Cert-Manager limitation 😁 . One should follow the "hacky" approach   because Cert-Manager itself does not allow setting two different encryption passwords. 

Having said that, I believe that even if it a limitation of Cert-Manager itself, having the ability to configure the keys for the provided secret may come handy (plus, would allow to skip the "hacky approach - though you'd still need to point **to the same key** if you are using Cert-Manager-). However, I would rename these two to something more tightly related to `passwordsSecretName` (just as we did with `existingSecret):

- `keystorePasswordSecretKey` --> `passwordsSecretKeystoreKey`
- `truststorePasswordSecretKey` --> `passwordsSecretTruststoreKey`

WDYT?

Reply:
> I am not really sure about this, in fact, I believe that this chart is setting two different random passwords if the TLS configuration is autogenerated instead of provided:

You are right, I read the documentation too fast, sorry. Zookeeper allows different password between **keystore** and **truststore** but not between **keystore** and **certificate key**.

> You are right! But then, I believe this is really a Cert-Manager limitation :grin: . One should follow the "hacky" approach because Cert-Manager itself does not allow setting two different encryption passwords.

I agree with this point too, this is a limitation from cert-manager currently...

But I think having the ability to configure the keys for the secret can be useful as you said (and we do not have to maintain a secret with 2 same keys... but yes, we need to point to the same key). I updated the parameters accordingly!

Reply:
Thanks!